profiler/profiler.go

```diff
@@ -154,6 +158,15 @@ func (p *profiler) run() {
 	if _, ok := p.cfg.types[BlockProfile]; ok {
 		runtime.SetBlockProfileRate(p.cfg.blockRate)
 	}
+	if s, ok := p.cfg.statsd.(statsdGauge); ok {
+		// If the statsd client supports gauges (e.g.
+		// github.com/datadog/datadog-go/v5/statsd) then we can send
+		// gauges reporting the current profiler configuration
```

Comment:
NIT: Maybe add a note here that these new metrics may be removed again in the future ... ? I'm okay with this having these as a stop-gap, but going forward we might want to move this into the telemetry pkg.